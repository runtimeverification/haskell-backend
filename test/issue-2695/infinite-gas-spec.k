module VERIFICATION
    imports INFINITE-GAS

    syntax KItem ::= runLemma ( Step ) | doneLemma ( Step )
 // -------------------------------------------------------
    rule <k> runLemma(S) => doneLemma(S) ... </k>

    syntax Step ::= Bool
 // --------------------

endmodule

module INFINITE-GAS-SPEC
    imports VERIFICATION

    claim <k> runLemma(#gas(GAVAIL) <Int log2Int(G0) +Int 1)  => doneLemma(false) ... </k> requires 0 <Int G0 andBool G0 <=Int 1

endmodule