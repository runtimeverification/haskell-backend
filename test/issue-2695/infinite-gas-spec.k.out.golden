  #Ceil ( log2Int ( G0 ) )
#And
  #Not ( {
    false
  #Equals
    #gas ( GAVAIL ) <Int log2Int ( G0 ) +Int 1
  } )
#And
  <k>
    doneLemma ( #gas ( GAVAIL ) <Int log2Int ( G0 ) +Int 1 ) ~> _DotVar1 ~> .
  </k>
#And
  {
    true
  #Equals
    0 <Int G0
  }
#And
  {
    true
  #Equals
    G0 <=Int 1
  }
